aea/test_tools/test_cases.py

```diff
@@ -0,0 +1,289 @@
+            vendor_name, item_type, item_name
+        )
+        self.run_cli_command("config", "set", json_path, False)
+
+    def disable_aea_logging(self):
+        """
+        Disable AEA logging of specific agent.
+        Run from agent's directory.
+
+        :return: None
+        """
+        config_update_dict = {
+            "agent.logging_config.disable_existing_loggers": "False",
+            "agent.logging_config.version": "1",
+        }
+        for path, value in config_update_dict.items():
+            self.run_cli_command("config", "set", path, value)
+
+    def run_cli_command(self, *args):
+        """
+        Run AEA CLI command.
+
+        :param args: CLI args
+
+        :return: None
+        """
+        result = self.runner.invoke(
+            cli, [*CLI_LOG_OPTION, *args], standalone_mode=False
+        )
+        if result.exit_code != 0:
+            raise AEATestingException(
+                "Failed to execute AEA CLI command with args {}.\n"
+                "Exit code: {}\nException: {}".format(
+                    args, result.exit_code, result.exception
+                )
+            )
+
+    def run_agent(self, *args):
+        """
+        Run agent as subprocess.
+        Run from agent's directory.
+
+        :param *args: CLI args
+
+        :return: subprocess object.
+        """
+        process = subprocess.Popen(  # nosec
+            [sys.executable, "-m", "aea.cli", "run", *args],
+            stdout=subprocess.PIPE,
+            env=os.environ.copy(),
+        )
+        return process
+
+    def initialize_aea(self, author=AUTHOR):
+        """
+        Initialize AEA locally with author name.
+
+        :return: None
+        """
+        self.run_cli_command("init", "--local", "--author", author)
+
+    def create_agents(self, *agents_names):
+        """
+        Create agents in current working directory.
+
+        :param *agents_names: str agent names.
+
+        :return: None
+        """
+        for name in agents_names:
+            self.run_cli_command("create", "--local", name, "--author", "fetchai")
+
+    def delete_agents(self, *agents_names):
+        """
+        Delete agents in current working directory.
+
+        :param *agents_names: str agent names.
+
+        :return: None
+        """
+        for name in agents_names:
+            self.run_cli_command("delete", name)
+
+    def add_item(self, item_type, public_id):
+        """
+        Add an item to the agent.
+        Run from agent's directory.
+
+        :param item_type: str item type.
+        :param item_type: str item type.
+
+        :return: None
+        """
+        self.run_cli_command("add", "--local", item_type, public_id)
+
+    def run_install(self):
+        """
+        Execute AEA CLI install command.
+        Run from agent's directory.
+
+        :return: None
+        """
+        self.run_cli_command("install")
+
+    @staticmethod
```

Comment:
This seems too specific for the generic test case class.

Reply:
184-272 seem specific to a derived test case class called `EchoTest` or something..not sure these methods are common enough to warrant inclusion here.